src/Microsoft.Azure.WebJobs.Host/InvokeFunctionFilterAttribute.cs

```diff
@@ -0,0 +1,124 @@
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = true, AllowMultiple = true)]
+    public sealed class InvokeFunctionFilterAttribute : InvocationFilterAttribute
+    {
+        /// <summary>
+        /// Constructs a new instance
+        /// </summary>
+        /// <param name="executingFilter">The name of the function to execute before the target function is called</param>
+        /// <param name="executedFilter">The name of the function to execute after the target function is called</param>
+        public InvokeFunctionFilterAttribute(string executingFilter = null, string executedFilter = null)
+        {
+            ExecutingFilter = executingFilter;
+            ExecutedFilter = executedFilter;
+        }
+
+        /// <summary>
+        /// The name of the function to execute before the target function is called
+        /// </summary>
+        public string ExecutingFilter { get; }
+
+        /// <summary>
+        /// The name of the function to execute after the target function is called
+        /// </summary>
+        public string ExecutedFilter { get; }
+
+        /// <inheritdoc/>
+        public override async Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            if (executingContext == null)
+            {
+                throw new ArgumentNullException(nameof(executingContext));
+            }
+
+            if (!string.IsNullOrEmpty(ExecutingFilter))
+            {
+                executingContext.Logger.LogInformation("Executing Function Filter '" + ExecutingFilter + "'");
+
+                await InvokeJobFunctionAsync(ExecutingFilter, executingContext, cancellationToken);
+            }
+            else
+            {
+                await base.OnExecutingAsync(executingContext, cancellationToken);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override async Task OnExecutedAsync(FunctionExecutedContext executedContext, CancellationToken cancellationToken)
+        {
+            if (executedContext == null)
+            {
+                throw new ArgumentNullException(nameof(executedContext));
+            }
+
+            if (!string.IsNullOrEmpty(ExecutedFilter))
+            {
+                executedContext.Logger.LogInformation($"Executing Function Filter '{ExecutedFilter}'");
+
+                await InvokeJobFunctionAsync(ExecutedFilter, executedContext, cancellationToken);
+            }
+            else
+            {
+                await base.OnExecutedAsync(executedContext, cancellationToken);
+            }
+        }
+
+        internal static async Task InvokeJobFunctionAsync<TContext>(string methodName, TContext context, CancellationToken cancellationToken) where TContext : FunctionInvocationContext
+        {
+            MethodInfo methodInfo = null;
+
+            foreach (var type in context.Config.TypeLocator.GetTypes())
```

Comment:
Can you make this an extension method on TypeLocator, something like

`static MethodInfo GetFunctionByName(this ITypeLocator typeLocator, string methodName)`

Also, assuming that the `methodName` is constant, I'd recommend caching the MethodInfo, ParameterInfo[], etc in the filter itself so that we don't run expensive reflection calls every execution.